cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/ArtifactLocalizerHttpHandlerInternal.java

```diff
@@ -49,10 +56,25 @@ public ArtifactLocalizerHttpHandlerInternal(ArtifactLocalizer artifactLocalizer)
   public void artifact(HttpRequest request, HttpResponder responder,
                        @PathParam("namespace-id") String namespaceId,
                        @PathParam("artifact-name") String artifactName,
-                       @PathParam("artifact-version") String artifactVersion) throws Exception {
+                       @PathParam("artifact-version") String artifactVersion) {
 
     ArtifactId artifactId = new ArtifactId(namespaceId, artifactName, artifactVersion);
-    File artifactPath = artifactLocalizer.getAndUnpackArtifact(artifactId);
-    responder.sendString(HttpResponseStatus.OK, artifactPath.toString());
+    try {
+      File artifactPath = artifactLocalizer.getAndUnpackArtifact(artifactId);
+      responder.sendString(HttpResponseStatus.OK, artifactPath.toString());
+    } catch (ArtifactNotFoundException ex) {
```

Comment:
Maybe check if the exception is an instance of http error status provider and the user the http status from that.

e.g.
      if ex instanceof HttpErrorStatusProvider {
      }

Reply:
Done, thanks!